public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -37,6 +41,97 @@ interface TimeAndValue {
   [TIME_SERIES_VALUE_FIELD_NAME]: number;
 }
 
+// V2 result trasnformer used to transform query results from queries that were run trough prometheus backend
+export function transformV2(response: DataQueryResponse, options: DataQueryRequest<PromQuery>) {
+  const promResults: DataFrame[] = cloneDeep(response.data);
+
+  // Get refIds that have table format as we need to process those to table reuslts
+  const tableRefIds = options.targets.map((target) => {
+    if (target.format === 'table') {
+      return target.refId;
+    }
+    return;
+  });
+
+  // For table results, we need to transform data frames to table data frames
+  const tableResults = promResults.filter((dataFrame) => tableRefIds.includes(dataFrame.refId));
```

Comment:
here you filter based on a function, and then lower you filter again with the opposite of that function.
`lodash` has a function for this called `partition`, it is like `array.filter`, but returns two arrays: one with items that did match, and one with items that did not match.

```suggestion
  const [tableResults, otherResults] = partition(promResults, (dataFrame) => tableRefIds.includes(dataFrame.refId));
```

Reply:
Updated in d7707c5